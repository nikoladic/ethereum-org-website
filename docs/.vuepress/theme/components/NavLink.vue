// Do not use classes in this file // Pass classes into this via NavLinks.vue
<template>
  <router-link
    ref="el"
    :to="link"
    :class="childClass"
    v-if="!isExternal(link)"
    :exact="exact"
    @click.native="$emit('nav-toggle', false)"
    >{{ item.text }}</router-link
  >
  <a
    v-else-if="!item.hideMobile"
    :href="link"
    :class="childClass + ' nav-link hide-icon'"
    v-on:click.native="$emit('nav-toggle', false)"
    :target="isMailto(link) || isTel(link) ? null : '_blank'"
    :rel="isMailto(link) || isTel(link) ? null : 'noopener noreferrer'"
  >
    {{ item.text }}
    <OutboundLink />
  </a>
</template>

<script>
import { isExternal, isMailto, isTel, ensureExt } from '../utils/util'

export default {
  props: {
    item: {
      required: true
    },
    childClass: {
      type: String,
      required: false
    },
    closeMenu: {
      type: Function
    }
  },
  computed: {
    link() {
      return ensureExt(this.item.link)
    },
    exact() {
      if (this.$site.locales) {
        return Object.keys(this.$site.locales).some(rootLink => {
          return rootLink === this.link
        })
      }
      return this.link === '/'
    }
  },

  methods: {
    isExternal,
    isMailto,
    isTel
  }
}
</script>
