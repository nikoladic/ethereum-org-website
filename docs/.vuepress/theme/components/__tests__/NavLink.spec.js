import { shallowMount } from '@vue/test-utils'
import NavLink from "@/theme/components/NavLink"

describe('NavLink', () => {
  test('render as external link', () => {
    const wrapper = shallowMount(NavLink, {
      propsData: {
        item: {
          text: "external link",
          link: "https://google.com"
        }
      },
      stubs: ['router-link', 'OutboundLink']
    })
    expect(wrapper.element).toMatchSnapshot()
  })

  test('render as email', () => {
    const wrapper = shallowMount(NavLink, {
      propsData: {
        item: {
          text: "email",
          link: "mailto:abc@google.com"
        }
      },
      stubs: ['router-link', 'OutboundLink']
    })
    expect(wrapper.element).toMatchSnapshot()
  })

  test('render as tel', () => {
    const wrapper = shallowMount(NavLink, {
      propsData: {
        item: {
          text: "tel",
          link: "tel:1778901652"
        }
      },
      stubs: ['router-link', 'OutboundLink']
    })
    expect(wrapper.element).toMatchSnapshot()
  })

  test('render as link with locales', () => {
    const wrapper = shallowMount(NavLink, {
      propsData: {
        item: {
          text: "English",
          link: "en"
        }
      },
      mocks: {
        $site: {
          locales: {
            "en": {
              label: "English",
              lang: "en"
            },
            "de": {
              label: "Deutsch",
              lang: "de"
            }
          },
        }
      },
      stubs: ['router-link', 'OutboundLink']
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
