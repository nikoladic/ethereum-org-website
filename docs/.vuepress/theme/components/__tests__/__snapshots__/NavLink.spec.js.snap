// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavLink render as email 1`] = `
<a
  class="nav-link external"
  href="mailto:abc@google.com"
>
  
  email
  
  <outboundlink-stub />
</a>
`;

exports[`NavLink render as external link 1`] = `
<a
  class="nav-link external"
  href="https://google.com"
  rel="noopener noreferrer"
  target="_blank"
>
  
  external link
  
  <outboundlink-stub />
</a>
`;

exports[`NavLink render as link with locales 1`] = `
<router-link-stub
  class="nav-link"
  to="/"
>
  English
</router-link-stub>
`;

exports[`NavLink render as tel 1`] = `
<a
  class="nav-link external"
  href="tel:1778901652"
>
  
  tel
  
  <outboundlink-stub />
</a>
`;
